lib/ansible/plugins/lookup/onepassword.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (c) 2018, Scott Buchanan <sbuchanan@ri.pn>
+# Copyright (c) 2016, Andrew Zenk <azenk@umn.edu> (lastpass.py used as starting point)
+# Copyright (c) 2018, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = """
+    lookup: onepassword
+    author:
+      -  Scott Buchanan <sbuchanan@ri.pn>
+      -  Andrew Zenk <azenk@umn.edu>
+    version_added: "2.6"
+    requirements:
+      - op (1Password command line utility: https://support.1password.com/command-line/)
+      - must have already logged into 1Password using op CLI
+    short_description: fetch data from 1Password
+    description:
+      - use the op command line utility to fetch specific fields from 1Password
+    options:
+      _terms:
+        description: identifier(s) of object(s) from which you want to retrieve the field (UUID, name or domain)
+        required: True
+      field:
+        description: field to return from 1Password
+        default: 'password'
+      vault:
+        description: vault to retrieve from (if absent will search all available vaults)
+        default: None
+"""
+
+EXAMPLES = """
+- name: "retrieve password from entry named KITT in any vault"
+  debug: password="{{ lookup('onepassword', 'KITT') }}"
+
+- name: "retrieve username from entry named HAL 9000 in vault Discovery"
+  debug: password="{{ lookup('onepassword', 'HAL 9000', field='username', vault='Discovery') }}"
+"""
+
+RETURN = """
+  _raw:
+    description: field data requested
+"""
+
+import json
+from subprocess import Popen, PIPE
+
+from ansible.errors import AnsibleError
+from ansible.plugins.lookup import LookupBase
+
+
+class OnePassException(AnsibleError):
+
+    pass
+
+
+class OnePass(object):
+
+    def __init__(self, path='op'):
+        self._cli_path = path
+
+    @property
+    def cli_path(self):
+        return self._cli_path
+
+    def assert_logged_in(self):
+        try:
+            self._run(["get", "account"])
+        except:
+            raise OnePassException("Not logged into 1Password: please run 'op signin' first")
+
+    def get_field(self, item_id, field, vault=None):
+        args = ["get", "item", item_id]
```

Comment:
I think it'd be worthwhile to make `item` the default but accept other subcommands as well. Supporting `get` as well as `list` would also be helpful.

Reply:
I could do that, but unless you think adding those other commands as high-level options to the plugin is a prerequisite for merging, I'd argue for a YAGNI approach: abstract it out when needed, not before. I could be persuaded, though, if you feel strongly.

Reply:
It's probably more of a want than a need at this point, so let's just start with what you have. I'm getting ahead with all the ways I can see this being potentially used.